src/metal/scheduler.ts

```diff
@@ -25,18 +25,38 @@ let tokenCounter = 0;
 export class Frame implements FrameInterface {
   constructor(
     public timestamp: number,
-    public scrollTop: number,
     public scrollLeft: number,
+    public scrollTop: number,
+    public x: number,
+    public y: number,
     public width: number,
     public height: number
   ) {}
-  static generate(): Frame {
+  static generate(root?: Element): Frame {
```

Comment:
This hotpath is already not great, needs work related to https://github.com/linkedin/spaniel/issues/12